File: src/main/java/me/dyzjct/kura/module/hud/huds/WaterMark.java
package me.dyzjct.kura.module.hud.huds;

import me.dyzjct.kura.module.HUDModule;
import me.dyzjct.kura.setting.FloatSetting;
import me.dyzjct.kura.setting.Setting;
import me.dyzjct.kura.setting.StringSetting;
import me.dyzjct.kura.utils.font.CFont;
import me.dyzjct.kura.utils.font.RFontRenderer;
import org.lwjgl.opengl.GL11;

import java.util.concurrent.CopyOnWriteArrayList;

@HUDModule.Info(name = "WaterMark", x = 20, y = 20)
public class WaterMark extends HUDModule {
    public CopyOnWriteArrayList<RFontRenderer> fonts = new CopyOnWriteArrayList<>();
    public StringSetting text = this.ssetting("ViewText", "Kura");
//    public ColorSetting color = csetting("Color", new Color(255,255,255));
    public Setting<Float> saturation = this.fsetting("Saturation",1.0f,0f,1.0f);
    public Setting<Float> brightness = this.fsetting("Brightness",1.0f,0f,1.0f);
    public Setting<Integer> alpha = isetting("Alpha",90,1,255);
    public FloatSetting speed = this.fsetting("RainbowSpeed", 1.0f, 0.0f, 1.0f);
    public FloatSetting Scala = this.fsetting("Scala", 1.0f, 0.0f, 3.0f);
    RFontRenderer font = new RFontRenderer(new CFont.CustomFont("/assets/fonts/font.ttf", 47.0f, 0), true, false);

    @Override
    public void onRender() {
        if (!fonts.contains(font)) {
            fonts.add(font);
        }
//        int fontColor = new Color(GuiManager.getINSTANCE().getRed() / 255f, GuiManager.getINSTANCE().getGreen() / 255f, GuiManager.getINSTANCE().getBlue() / 255f, 1F).getRGB();

        fonts.forEach(f -> {
            GL11.glPushMatrix();
            GL11.glTranslated(this.x, (float) this.y, 0);
            GL11.glScaled((double) this.Scala.getValue(), (double) this.Scala.getValue(), 0.0);
            f.drawString(this.text.getValue(), 0, 0, speed.getValue().floatValue(), saturation.getValue().floatValue(), this.brightness.getValue().floatValue(), 50, this.alpha.getValue(),false);
            GL11.glPopMatrix();
            this.width = (int) ((float) f.getStringWidth(this.text.getValue()) * this.Scala.getValue());
            this.height = (int) ((float) f.getHeight() * this.Scala.getValue());
        });
    }
    public enum Mode {
        Rainbow , Custom;
    }
}

File: src/main/java/me/dyzjct/kura/module/hud/huds/WaterMark.kt
package me.dyzjct.kura.module.hud.huds

import me.dyzjct.kura.manager.FontManager
import me.dyzjct.kura.manager.GuiManager
import me.dyzjct.kura.module.HUDModule
import me.dyzjct.kura.setting.Setting
import me.dyzjct.kura.utils.color.ColorUtil
import org.lwjgl.opengl.GL11
import java.awt.Color

/**
 * Created by dyzjct on 29/12/2022
 */
@HUDModule.Info(name = "WaterMark", x = 20, y = 20, width = 50, height = 20)
class WaterMark : HUDModule() {
    private var text = ssetting("ViewText", "kura")
    private var colormod = msetting("ColorMod", Welcomer.ColorMode.GuiSync)
    private var color = csetting("Color", Color(255, 255, 255))


    fun generateColor(): Int {
        val fontColor = Color(
            GuiManager.getINSTANCE().red / 255f,
            GuiManager.getINSTANCE().green / 255f,
            GuiManager.getINSTANCE().blue / 255f,
            1f
        ).rgb
        val custom = Color(color.value.red, color.value.green, color.value.blue).rgb
        when (colormod.value) {
            Welcomer.ColorMode.Rainbow -> return ColorUtil.staticRainbow().rgb
            Welcomer.ColorMode.GuiSync -> return fontColor
            Welcomer.ColorMode.Custom -> return custom
        }
        return -1
    }

    override fun onRender() {
        GL11.glPushMatrix()
        GL11.glTranslated(x.toDouble(), y.toFloat().toDouble(), 0.0)
//        GL11.glScaled(Scala.value.toDouble(), Scala.value.toDouble(), 0.0)
        FontManager.font2!!.drawString(text.value, 0.0, 0.0, generateColor(), false)
        GL11.glPopMatrix()
//        width = (FontManager.font2!!.getStringWidth(text.value).toFloat() * Scala.value).toInt()
//        height = (FontManager.font2!!.height.toFloat() * Scala.value).toInt()
    }

    enum class ColorMode {
        Rainbow, Custom, GuiSync
    }
}